clang/test/CIR/CodeGen/static-vars.c

```diff
@@ -0,0 +1,37 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-cir %s -o %t.cir
```

Comment:
How about the cpp tests? In this case I'd prefer if C and C++ are tested in different files. If you added CIRGen for cpp it should be all tested as part of this PR, otherwise paths should be asserted and supported added in a follow up PR. Specially in light of recent static discussions, it's important that we test every bit for every added path, as to not miss any details.

Reply:
Added C++ tests for function static vars too.